vm/src/pyobject.rs

```diff
@@ -1599,6 +1625,33 @@ pub trait PyValue: fmt::Debug + Sized + 'static {
     }
 }
 
+pub trait PyImmutableClass: PyValue {
```

Comment:
What's the reason for this being a separate trait from `PyValue`?

Reply:
My initial thought was that as PyValue::class required the VM, then not all PyValue types are immutable/can be shared between VMs. But depending on what we want the VM/PyContext distinction to be, these methods might move into PyValue.

Reply:
I think they should be part of `PyValue` either way. From the point of view of the VM, they are all "immutable" (i.e. only accessible via shared refs).

Reply:
If I move 
```rust
fn create_type(ctx: &PyContext) -> PyClassRef;
```
Into PyValue then how is that going to work for Regex? As it requires the VM to get the type.

Reply:
It currently uses the vm, but wouldn't everything be moved to use these `get_type` / `set_type` methods you added to `PyContext`? I don't see any reason to have two different ways to register built-in types.